src/iqm/qiskit_iqm/iqm_backend.py

```diff
@@ -44,28 +46,36 @@ def get_num_or_zero(name: str) -> int:
             return int(match.group(1)) if match else 0
 
         qb_to_idx = {qb: idx for idx, qb in enumerate(sorted(architecture.qubits, key=get_num_or_zero))}
+        operations = architecture.operations
 
         target = Target()
         # There is no dedicated direct way of setting just the qubit connectivity and the native gates to the target.
         # Such info is automatically deduced once all instruction properties are set. Currently, we do not retrieve
         # any properties from the server, and we are interested only in letting the target know what is the native gate
         # set and the connectivity of the device under use. Thus, we populate the target with None properties.
-        target.add_instruction(
-            RGate(Parameter('theta'), Parameter('phi')), {(qb_to_idx[qb],): None for qb in architecture.qubits}
-        )
+        if 'prx' in operations:
+            target.add_instruction(
+                RGate(Parameter('theta'), Parameter('phi')), {(qb_to_idx[qb],): None for [qb] in operations['prx']}
+            )
         target.add_instruction(IGate(), {(qb_to_idx[qb],): None for qb in architecture.qubits})
         # Even though CZ is a symmetric gate, we still need to add properties for both directions. This is because
         # coupling maps in Qiskit are directed graphs and the gate symmetry is not implicitly planted there. It should
         # be explicitly supplied. This allows Qiskit to have coupling maps with non-symmetric gates like cx.
-        target.add_instruction(
-            CZGate(),
-            {
-                (qb_to_idx[qb1], qb_to_idx[qb2]): None
-                for pair in architecture.qubit_connectivity
-                for qb1, qb2 in (pair, pair[::-1])
-            },
-        )
-        target.add_instruction(Measure(), {(qb_to_idx[qb],): None for qb in architecture.qubits})
+        if 'cz' in operations:
+            target.add_instruction(
+                CZGate(),
+                {
+                    (qb_to_idx[qb1], qb_to_idx[qb2]): None
```

Comment:
This is implicitly duplicated knowledge which operation is sensitive to the order of qubits and which is not, `is_directed_operation` should also be used.

Reply:
I've now generalized qiskit target configuration (how to define connectivity) and this info is now queried from IQM client.